src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -4365,13 +4365,20 @@ final void processEnvChange(TDSReader tdsReader) throws SQLServerException {
                     }
                 }
 
-                isRoutedInCurrentAttempt = true;
-                routingInfo = new ServerPortPlaceHolder(routingServerName, routingPortNumber, null, integratedSecurity);
+                int px = routingServerName.indexOf('\\');
```

Comment:
This will be checked again when we call ServerportPlaceHolder later. The routingInstanceName doesn't seem necessary which can be set up in `ServerPortPlaceHolder` constructor.